packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -811,6 +841,27 @@ def nb_participants(self) -> int:
         participants = cast(List, self.db.get("participants", []))
         return len(participants)
 
+    def initialize(self, synced_setup_data: Dict[str, List[Any]]) -> None:
+        """Initialize the synchronized data.
+
+        This method should only be used to initialize the data, once,
+        and only after the agents have reached consensus about the initial setup data.
+        `RegistrationStartupRound` is meant to utilize this method.
+
+        :param synced_setup_data: the synchronized setup data to initialize the database with.
+        """
+        self.db.initialize(synced_setup_data)
+        missing_keys = []
+        for mandatory_key in self.default_db_keys:
+            # we do not check if the attribute exists since this is done during the chaining
+            if not getattr(self, mandatory_key):
+                missing_keys.append(mandatory_key)
+
+        enforce(
+            not missing_keys,
+            f"There are missing setup data {missing_keys} after initialization!",
+        )
```

Comment:
We enforce the `default_db_keys` to be non-null after synchronizing the setup data.